cmd/relayer/main.go

```diff
@@ -0,0 +1,426 @@
+// Package main is the entrypoint for the swap-specific Ethereum transaction relayer.
+// It's purpose is to allow swaps users (ETH-takers in particular) to submit calls to the
+// swap contract to claim their ETH from an account that does not have any ETH in it.
+// This improves the swap UX by allowing users to obtain ETH without already having any.
+// In this case, the relayer submits the transaction on the user's behalf, paying their
+// gas fees, and (optionally) receiving a small percentage of the swap's value as payment.
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/cockroachdb/apd/v3"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	rcontracts "github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	net "github.com/athanorlabs/go-relayer/net"
+	"github.com/athanorlabs/go-relayer/relayer"
+	rrpc "github.com/athanorlabs/go-relayer/rpc"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	swapnet "github.com/athanorlabs/atomic-swap/net"
+
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	flagDataDir          = "data-dir"
+	flagEndpoint         = "endpoint"
+	flagForwarderAddress = "forwarder-address"
+	flagKey              = "key"
+	flagRPC              = "rpc"
+	flagRPCPort          = "rpc-port"
+	flagDeploy           = "deploy"
+	flagLog              = "log-level"
+	// TODO: do we need this, or can we assume all swap relayers will need to be on the p2p network?
+	flagWithNetwork       = "with-network"
+	flagLibp2pKey         = "libp2p-key"
+	flagLibp2pPort        = "libp2p-port"
+	flagBootnodes         = "bootnodes"
+	flagRelayerCommission = "relayer-commission"
+
+	defaultLibp2pPort = 10900
+)
+
+var (
+	log = logging.Logger("main")
+
+	flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:  flagDataDir,
+			Usage: "Path to store swap artifacts",
+			Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
+		},
+		&cli.StringFlag{
+			Name:  flagEndpoint,
+			Value: "http://localhost:8545",
+			Usage: "Ethereum RPC endpoint",
+		},
+		&cli.StringFlag{
+			Name:  flagKey,
+			Value: "{HOME}/.atomicswap/{ENV}/relayer/eth.key",
```

Comment:
Can you update  `docs/default-file-locations.md` with all the new relayer files? For this entry, we should have a note that the wallet key must be different from `swapd`'s ETH wallet key.

Reply:
should the default value be `{DATA-DIR}/relayer` actually? for some reason, `{DATA-DIR}` didn't seem to be properly recognized when I tried running this locally

Reply:
The values in braces (`{...}`) are like that for the `--help`. The true default value can depend on other flags, so we don't try to resolve it for the purposes of the help. When you later check the value, use `IsSet` to detect when a user has set the flag instead of using `String`. If it is not set, then you resolve the data-dir value and create the path.

Reply:
updated